File: backend/app/schemas/social.py
"""Social (L2) API schemas — the Taste Circle: following + the explore feed."""
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.collection import CollectionResponse


class UserPublic(BaseModel):
    """Public view of a user — no email or private fields."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    username: str
    display_name: str | None = None


class FollowRequest(BaseModel):
    followee_id: uuid.UUID


class FollowStatusResponse(BaseModel):
    """Whether the current user follows `user`, used on profiles."""

    user: UserPublic
    is_following: bool


class FeedItem(BaseModel):
    """One public collection surfaced from someone you follow."""

    owner: UserPublic
    collection: CollectionResponse
    created_at: datetime

File: packages/types/src/social.ts
// Mirrors backend app/schemas/social.py — the L2 Taste Circle.

import type { Collection } from "./collection";

// Public view of a user (no email / private fields).
export interface UserPublic {
  id: string;
  username: string;
  display_name: string | null;
}

export interface FollowStatus {
  user: UserPublic;
  is_following: boolean;
}

// One public collection surfaced from someone you follow.
export interface FeedItem {
  owner: UserPublic;
  collection: Collection;
  created_at: string;
}
